core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartUpscaleGetRecoveryCandidatesHandler.java

```diff
@@ -0,0 +1,187 @@
+import com.sequenceiq.cloudbreak.cmtemplate.CMRepositoryVersionUtil;
+import com.sequenceiq.cloudbreak.converter.CloudInstanceIdToInstanceMetaDataConverter;
+import com.sequenceiq.cloudbreak.converter.spi.InstanceMetaDataToCloudInstanceConverter;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.stack.InstanceMetaDataService;
+import com.sequenceiq.cloudbreak.service.stack.RuntimeVersionService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.view.InstanceMetadataView;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StopStartUpscaleGetRecoveryCandidatesHandler implements CloudPlatformEventHandler<StopStartUpscaleGetRecoveryCandidatesRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StopStartUpscaleGetRecoveryCandidatesHandler.class);
+
+    @Inject
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private RuntimeVersionService runtimeVersionService;
+
+    @Inject
+    private CloudInstanceIdToInstanceMetaDataConverter cloudInstanceIdToInstanceMetaDataConverter;
+
+    @Inject
+    private InstanceMetaDataToCloudInstanceConverter instanceMetaDataToCloudInstanceConverter;
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public Class<StopStartUpscaleGetRecoveryCandidatesRequest> type() {
+        return StopStartUpscaleGetRecoveryCandidatesRequest.class;
+    }
+
+    @Override
+    public void accept(Event<StopStartUpscaleGetRecoveryCandidatesRequest> event) {
+        StopStartUpscaleGetRecoveryCandidatesRequest request = event.getData();
+        LOGGER.info("StopStartUpscaleGetRecoveryCandidatesHandler: {}", event.getData().getResourceId());
+        if (!request.isFailureRecoveryEnabled()) {
+            StopStartUpscaleGetRecoveryCandidatesResult result =  new StopStartUpscaleGetRecoveryCandidatesResult(request.getResourceId(), request,
+                    Collections.emptyList(), request.getAdjustment(), request.getHostGroupName());
+            notify(result, event);
+            return;
+        }
+
+        CloudContext cloudContext = request.getCloudContext();
+        try {
+            CloudConnector connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            Stack stack = stackService.getByIdWithLists(request.getResourceId());
```

Comment:
Could you please change to StackDto.getById(...)

Reply:
Addressed this and below comment